### ISSUE TITLE
Unexpected Stencil warning: "The state/prop "step" changed during rendering"

### ISSUE THREAD

**Issue by eriklharper** [ionic-team/stencil#2814]
**Stencil version:**

```
 @stencil/core@2.4.0
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**

Console warning below appears with the code in the example repo.

![image](https://user-images.githubusercontent.com/821864/107080489-35379800-67a6-11eb-87cd-73b75a67cb51.png)

**Expected behavior:**

No console warning should appear because `number` is a valid type to supply to the component's `step` property.

**Steps to reproduce:**

1. Clone, `npm i && npm start` this code to see the error:

**Related code:**
https://github.com/eriklharper/stencil-prop-changed-during-rendering

```tsx
import { Component, Host, h, Prop } from '@stencil/core';

@Component({
  tag: 'my-input',
  shadow: true,
})
export class MyInput {
  @Prop({ reflect: true }) step?: number | "any";
  render() {
    return (
      <Host>
        <input step={this.step}></input>
      </Host>
    );
  }
}
```
This seems to be related to the `number | "any"` typing for this property.  `"any"` is a valid value for html input's step attribute.  If you remove the `| "any"` from the type, the error goes away.

**Other information:**

Here is the original Slack thread I created to ask for help on this problem: https://stencil-worldwide.slack.com/archives/C79EANFL7/p1612383253310800

**Comment by tricki** [comment]
My guess is the problem is in [`parsePropertyValue`](https://github.com/ionic-team/stencil/blob/12fec21b2eb270cf2a8c30fa9deb09e8a49da5fd/src/runtime/parse-property-value.ts#L4) where `BUILD.propNumber && propType & MEMBER_FLAGS.Number` returns `false` because of the `| "any"` type.

Maybe that could be extended to check if `number` is a valid type for the property (even if not the only one) and then checking if it's a numeric string.

**Comment by rubenprins** [comment]
Same thing happens when using a type union like `number | 'other' | 'value'`.

**Comment by eriklharper** [comment]
This is still an issue within our input component that needs to support the `any` value for the step attribute.  It clogs up our console constantly in both development and in our test runner.  Really annoying.

**Comment by rwaskiewicz** [comment]
This should be fixed by https://github.com/ionic-team/stencil/pull/3106, which is a part of [Stencil v2.10.0](https://github.com/ionic-team/stencil/releases/tag/v2.10.0), which was released earlier today. I'm going to close this issue, please let us know if this release doesn't resolve the issue!

**Comment by eriklharper** [comment]
> This should be fixed by #3106, which is a part of [Stencil v2.10.0](https://github.com/ionic-team/stencil/releases/tag/v2.10.0), which was released earlier today. I'm going to close this issue, please let us know if this release doesn't resolve the issue!

This does appear to be fixed in my repro case.  Thanks Ryan @rwaskiewicz !
